packages/wrangler/src/init.ts

```diff
@@ -1035,6 +1052,18 @@ export function mapBindings(bindings: WorkerMetadataBinding[]): RawConfig {
 							};
 						}
 						break;
+					case "d1":
+						{
+							configObj.d1_databases = [
+								...(configObj.d1_databases ?? []),
+								{
+									binding: binding.name,
+									database_id: binding.id,
+									database_name: d1BindingsWithInfo[binding.id].name,
```

Comment:
You might be tempted to ask why we don't just teach EWC about `database_name` - the thing is, EWC doesn't store or care about the name of the DB, only D1's API knows it, so we need to fetch it if we want Workers initialised with `--from-dash` to be capable of running `wrangler d1` commands.